src/components/widgets/RadioWidget.js

```diff
@@ -32,7 +31,7 @@ function RadioWidget(props) {
             <input
               type="radio"
               checked={checked}
-              name={name}
+              name={idToPath(id)}
```

Comment:
@travisdahl this is a backwards-incompatible change; why do you think `idToPath(id)` is more appropriate in this case rather than the previously randomly-generated field name?

Reply:
Is this really not backwards compatible? I don't think anyone can rely on using the random generated name in their code in some way. We we're thinking that the name only needed to be unique. It should still be unique in this case because the paths are unique for the form. It also seemed to make more sense to have the name be related to the form/field rather than a random name, more in line with a traditional html form.

Reply:
That makes sense; this is definitely an improvement over the randomly-generated name.

My only concern is that now the name doesn't reflect the data path anymore. Let's say I select "foo" and "fuzz" in the playground:
![image](https://user-images.githubusercontent.com/1689183/53293083-2bb3f180-3782-11e9-865d-ef6078cf0e72.png)

Then the checkboxes with names "multipleChoicesList.0" and "multipleChoicesList.2" are checked.

Contrast this with if I didn't use a checkbox widget:
![image](https://user-images.githubusercontent.com/1689183/53293098-59993600-3782-11e9-98b1-58b8f68d0a89.png)

In this case, the `<select>` elements with names "multipleChoicesList.0" and "multipleChoicesList.1" are the ones having the values.

I would think that the latter is more natural, as, really, the value of `multipleChoicesList` is an array with two elements, and having names "multipleChoicesList.0" and "multipleChoicesList.1" would reflect that.

But maybe I'm just worried about a non-issue -- do you think the concern I raised might not really practically matter?

Reply:
Good observation. You are correct that it doesn't reflect the data path anymore, but instead it really represents its location on the form. In practicality, it does not matter for the result of `omitUnusedData` because for the arrays fields, only the selected/present values will be in the formData and all of the array paths are present in the `event.target.elements`, so all of the array values will be picked.